### PR TITLE
Implement client specific attrs registration

### PR DESCRIPTION
**Pull request by dstansby**
Another part of #3770

### CODE REVIEW

File: changelog/3770.feature.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Fido clients (subclasses of `sunpy.net.base_client.BaseClient`) can now register their own attrs modules with `sunpy.net.attrs`. This allows clients which require attr classes specific to that client to register modules that can be used by the user i.e. ``a.vso``. It also allows clients implemented externally to sunpy to register attrs.
```
Comment (dstansby) [inline changelog/3770.feature.rst]:
```suggestion
Fido clients (subclasses of `sunpy.net.base_client.BaseClient`) can now register their own attrs modules with `sunpy.net.attrs`. This allows clients which require attr classes specific to that client to register attributes that can be used by the user i.e. ``a.vso``. It also allows clients implemented externally to sunpy to register attrs.
```
... unless I'm reading something wrong?

File: sunpy/net/dataretriever/sources/goes.py [hunk 2]
```diff
@@ -191,6 +191,10 @@ def _can_handle_query(cls, *query):
                 return all(chklist)
         return False
 
+    @classmethod
+    def _attrs_module(cls):
```
Comment (dstansby) [inline sunpy/net/dataretriever/sources/goes.py]: There should probably be some docs on how to write this function (ie. should be called `_attrs_module()`, should return these values...) for people wanting to do this themselves?

Comment (nabobalis) [reply sunpy/net/dataretriever/sources/goes.py]: I believe there is a seperate PR to update the docs.

File: changelog/3770.feature.rst (hunk 1 above)
Comment (Cadair) [reply changelog/3770.feature.rst]: You register a module (i.e. `a.vso` maps to `sunpy.net.vso.attrs`) which contains attrs (but is just a module). [here](https://github.com/sunpy/sunpy/pull/3869/files#diff-85c0984f67d7abb954d0f4a8d56fcb4fR46)

**Comment by Cadair** [comment]
@dstansby you happy with this?

**Pull request merged**